agent/agent.go

```diff
@@ -766,6 +766,9 @@ func (a *Agent) buildExternalGRPCServer() {
 	if a.config.HTTPSPort > 0 {
 		tls = a.tlsConfigurator
 	}
+	if a.tlsConfigurator != nil && a.tlsConfigurator.GRPCTLSConfigured() {
+		tls = a.tlsConfigurator
+	}
```

Comment:
⚠️ This could make upgrading client agents difficult because in many cases it'll (unexpectedly?) enable TLS on the gRPC port that Envoy is using for xDS.

This is because `GRPCTLSConfigured` is true in _any_ of the following cases:

- `tls.grpc.{cert_file,key_file}` are set
- `tls.defaults.{cert_file,key_file}` are set
- Auto-config or auto-encrypt are in use

Which is why we kept the https port/grpc coupling when introducing the new `tls` stanza, even though it doesn't make much sense now that TLS is configured separately.

If we're sure we want to go ahead with this change, we should probably just remove the above check (`if a.config.HTTPSPort > 0`) and always pass `tls`, because we already check `GRPCTLSConfigured` in `external.NewServer`.

Reply:
FWIW, I would be open to removing configuring TLS being dependent on HTTPs enablement.

In 1.12 we had to do differently to allow for an upgrade path to work at all.

For 1.13 you can just go through 1.12, fix up your settings and then move along to 1.13.

So for me, I saw we rip out the HTTPSPort check, always pass the tls configurator and add to the upgrade_specific.mdx with the details about the breaking change.

Reply:
That's a great point about there being an upgrade path through 1.12 now. I'm in favor of dropping it too!